* Pass the directory of the vorbis headers to cmake via CMAKE_REQUIRED_INCLUDES
* OGGVORBIS_VERSION is not defined

--- plugins/codec_ogg/CMakeLists.txt.orig	2017-06-05 19:06:34 UTC
+++ plugins/codec_ogg/CMakeLists.txt
@@ -68,10 +68,12 @@ IF (WITH_OGG_OPUS OR WITH_OGG_VORBIS)
 ### check for Vorbis headers and library                                  ###
 
     IF (WITH_OGG_VORBIS)
-	PKG_CHECK_MODULES(VORBIS REQUIRED vorbis>=1.0.0)
+	PKG_CHECK_MODULES(VORBIS REQUIRED vorbis>=1.0.0 IMPORTED_TARGET)
 	MESSAGE(STATUS "  Found vorbis codec library in ${VORBIS_LIBDIR}")
 	MESSAGE(STATUS "  Found vorbis codec headers in ${VORBIS_INCLUDEDIR}")
-	CHECK_LIBRARY_EXISTS(vorbis vorbis_bitrate_addblock "${VORBIS_INCLUDEDIR}" HAVE_LIBVORBISENC_V2)
+	set(CMAKE_REQUIRED_INCLUDES ${VORBIS_INCLUDEDIR})
+	set(CMAKE_REQUIRED_LIBRARIES PkgConfig::VORBIS)
+	CHECK_SYMBOL_EXISTS(vorbis_bitrate_addblock "vorbis/codec.h" HAVE_LIBVORBISENC_V2)
 	IF (NOT HAVE_LIBVORBISENC_V2)
 	    MESSAGE(FATAL_ERROR "libvorbis lacks the function vorbis_bitrate_addblock()")
 	ENDIF (NOT HAVE_LIBVORBISENC_V2)
@@ -81,6 +83,7 @@ IF (WITH_OGG_OPUS OR WITH_OGG_VORBIS)
 	MESSAGE(STATUS "  Found vorbisenc library in ${VORBISENC_LIBDIR}")
 	MESSAGE(STATUS "  Found vorbisenc headers in ${VORBISENC_INCLUDEDIR}")
 	SET(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} vorbisenc)
+        set(CMAKE_REQUIRED_INCLUDES ${VORBIS_INCLUDEDIR})
 
 	CHECK_INCLUDE_FILES("vorbis/codec.h;vorbis/vorbisenc.h" HAVE_OGG_VORBIS_HEADERS)
 	IF (NOT HAVE_OGG_VORBIS_HEADERS)
